src/pages/sales/SalesOrderDetails.tsx

```diff
@@ -68,56 +76,129 @@ const steps = [
+                  removeItemFromList(params.row.productName);
+                }}
+              >
+                Remove Item
+              </Button>
+            </>
+          );
+        }
+      }
+    }
+  ];
 
   useEffect(() => {
     setLoading(true);
-    const saleOrd = salesOrders.find((order) => {
-      return order.id === parseInt(id!);
-    });
+    id &&
+      asyncFetchCallback(
+        getSalesOrderDetailsSvc(id),
+        (salesOrder: SalesOrder) => {
+          if (salesOrder) {
+            setSalesOrder(salesOrder);
+            setEditSalesOrderItems([...salesOrder.salesOrderItems]);
+            setAvailProducts(
+              products.filter((product) => {
+                return salesOrder?.salesOrderItems.find(
+                  (orderItem) => orderItem.productName !== product.name
+                );
+              })
+            );
+            switch (salesOrder.orderStatus) {
+              case OrderStatus.CREATED: {
+                setActiveStep(0);
+                break;
+              }
+              case OrderStatus.PAID: {
+                setActiveStep(1);
+                break;
+              }
+              case OrderStatus.PREPARING: {
+                setActiveStep(2);
 
-    if (saleOrd) {
-      setSalesOrder(saleOrd);
-      switch (saleOrd.orderStatus) {
-        case OrderStatus.CREATED: {
-          setActiveStep(0);
-          break;
-        }
-        case OrderStatus.PAID: {
-          setActiveStep(1);
-          break;
-        }
-        case OrderStatus.PREPARING: {
-          setActiveStep(2);
-          break;
-        }
-        case OrderStatus.PREPARED: {
-          setActiveStep(3);
-          break;
-        }
-        case OrderStatus.SHIPPED: {
-          setActiveStep(4);
-          break;
-        }
-        case OrderStatus.COMPLETED: {
-          setActiveStep(5);
-          break;
-        }
-        default: {
-          break;
+                break;
+              }
+              case OrderStatus.PREPARED: {
+                setActiveStep(3);
+                break;
+              }
+              case OrderStatus.SHIPPED: {
+                setActiveStep(4);
+                break;
+              }
+              case OrderStatus.COMPLETED: {
+                setActiveStep(5);
+                break;
+              }
+              default: {
+                break;
+              }
+            }
+            setLoading(false);
+          } else {
+            setAlert({
+              severity: 'error',
+              message:
+                'Sales Order does nto exist. You will be redirected back to the Sales Order Overview page.'
+            });
+            setLoading(false);
+            setTimeout(() => navigate('/allSalesOrders'), 3500);
+          }
         }
-      }
-      setLoading(false);
-    }
-  }, [salesOrders]);
+      );
+  }, [id, navigate, products]);
 
   const nextStep = () => {
     switch (salesOrder?.orderStatus) {
```

Comment:
this switch statement also can be refactored
```suggestion
    if (activeStep < steps.length - 1) {
      setActiveStep(prev => prev + 1);
      const newStatus = steps[activeStep + 1].value;
      if (newStatus === OrderStatus.PREPARED) {
        setSalesOrder(order => order && {...order, orderStatus: newStatus, salesOrderItems: editSalesOrderItems });
        updateSalesOrder();
      } else {
        setSalesOrder(order => order && {...order, orderStatus: newStatus });
      }
    }
```